modules/pxp-module-puppet

```diff
@@ -53,11 +77,20 @@ def disabled?(run_result)
   !!(run_result =~ /disabled(.*?)Use 'puppet agent --enable' to re-enable/m)
 end
 
-def make_command_string(config, params)
-  env = params["env"].join(" ")
-  flags = params["flags"].join(" ")
+def make_environment_map(params)
+  env_dict = {}
+  params["env"].each do |entry|
+    key, value = entry.split('=')
+    env_dict[escape_string(key)] = escape_string(value)
```

Comment:
Same as the other comment.  As the API is taking a Hash here, surely that API should ensure the keys/values are passed through in a safe manner.